jenkins/jobs-templates/generate_jobs.py

```diff
@@ -0,0 +1,148 @@
+    'trigger-libs2ecore': {
+        'repo': 'https://github.com/S2E/libs2ecore.git',
+        'url': 'https://github.com/S2E/libs2ecore/'
+    },
+    'trigger-libs2eplugins': {
+        'repo': 'https://github.com/S2E/libs2eplugins.git',
+        'url': 'https://github.com/S2E/libs2eplugins/'
+    },
+    'trigger-libvmi': {
+        'repo': 'https://github.com/S2E/libvmi.git',
+        'url': 'https://github.com/S2E/libvmi/'
+    },
+    'trigger-guest-tools': {
+        'repo': 'https://github.com/S2E/guest-tools.git',
+        'url': 'https://github.com/S2E/guest-tools/'
+    },
+    'trigger-libfsigcxx': {
+        'repo': 'https://github.com/S2E/libfsigcxx.git',
+        'url': 'https://github.com/S2E/libfsigcxx/'
+    },
+    'trigger-tools': {
+        'repo': 'https://github.com/S2E/tools.git',
+        'url': 'https://github.com/S2E/tools/'
+    },
+    'trigger-klee': {
+        'repo': 'https://github.com/S2E/klee.git',
+        'url': 'https://github.com/S2E/klee/'
+    },
+    'trigger-libcoroutine': {
+        'repo': 'https://github.com/S2E/libcoroutine.git',
+        'url': 'https://github.com/S2E/libcoroutine/'
+    },
+    'trigger-libq': {
+        'repo': 'https://github.com/S2E/libq.git',
+        'url': 'https://github.com/S2E/libq/'
+    },
+}
+
+
+def gen_triggers(template_file, jobs_dir):
+    """
+    Generate one repository monitoring job per S2E repository.
+    A job can monitor only one repository, so we need to create one job for each repository.
+    This job will trigger other downstream jobs that will do the actual work
+    (e.g., checking code style, building, etc)
+    """
+    for job_name, env in TRIGGERS_CFG.iteritems():
+        with open(template_file, 'r') as fp:
+            template = Template(fp.read())
+            rendered = template.render(env)
+
+        job_dir = os.path.join(jobs_dir, job_name)
+        if not os.path.isdir(job_dir):
+            os.makedirs(job_dir, 0770)
+
+        job_file = os.path.join(job_dir, 'config.xml')
+        log.info('Rendering %s', job_file)
+        with open(job_file, 'w') as fp:
+            fp.write(rendered)
+
+
+def copy_cfg_file(job_cfg_file, jobs_dir, job_name):
+    job_dir = os.path.join(jobs_dir, job_name)
+    if not os.path.isdir(job_dir):
+        os.makedirs(job_dir, 0770)
+
+    job_file = os.path.join(job_dir, 'config.xml')
+
+    shutil.copyfile(job_cfg_file, job_file)
+
+
+def main():
+    parser = argparse.ArgumentParser(description='Generate Jenkins jobs from templates')
+    parser.add_argument('-j', '--jobs-dir', required=True,  help='Path to Jenkins jobs directory')
+    parser.add_argument('-t', '--templates-dir', required=True, help='Path to the templates directory')
+    args = parser.parse_args()
+
+    if not os.path.isdir(args.jobs_dir):
+        log.error('%s does not exist', args.jobs_dir)
+        exit(1)
+
+    if not os.path.isdir(args.templates_dir):
+        log.error('%s does not exist', args.templates_dir)
+        exit(1)
```

Comment:
Does this work? Isn’t `exit` provided by the `sys` module?

I.e., `sys.exit(1)`

Reply:
Yes, but apparently it should not be used, so I'll fix it. Pylint didn't complain.
https://stackoverflow.com/questions/19747371/python-exit-commands-why-so-many-and-when-should-each-be-used